bib/cmd/bootc-image-builder/image.go

```diff
@@ -77,21 +77,113 @@ func Manifest(c *ManifestConfig) (*manifest.Manifest, error) {
+		// allow all existing mountpoints to support size customizations
+		"/":     {Deny: false, Exact: true},
+		"/boot": {Deny: false, Exact: true},
+	})
+)
+
+func checkMountpoints(filesystems []blueprint.FilesystemCustomization, policy *pathpolicy.PathTrie) error {
+	errs := []error{}
+	for _, fs := range filesystems {
+		if err := policy.Check(fs.Mountpoint); err != nil {
+			errs = append(errs, err)
+		}
+		if fs.Mountpoint == "/var" {
+			// this error message is consistent with the errors returned by policy.Check()
+			// TODO: remove trailing space inside the quoted path when the function is fixed in osbuild/images.
+			errs = append(errs, fmt.Errorf("path '/var ' is not allowed"))
+		}
+	}
+	if len(errs) > 0 {
+		return fmt.Errorf("The following errors occurred while validating custom mountpoints:\n%w", errors.Join(errs...))
+	}
+	return nil
+}
+
+func checkFilesystemCustomizations(fsCustomizations []blueprint.FilesystemCustomization, ptmode disk.PartitioningMode) error {
+	var policy *pathpolicy.PathTrie
+	switch ptmode {
+	case disk.BtrfsPartitioningMode:
+		// btrfs subvolumes are not supported at build time yet, so we only
+		// allow / and /boot to be customized when building a btrfs disk (the
+		// minimal policy)
+		policy = mountpointMinimalPolicy
+	default:
+		policy = mountpointPolicy
+	}
+	if err := checkMountpoints(fsCustomizations, policy); err != nil {
 		return err
 	}
-	if customFS := customizations.GetFilesystems(); len(customFS) != 0 {
-		allowedMounts := []string{"/", "/boot"}
-		for _, mp := range customFS {
-			if !slices.Contains(allowedMounts, mp.Mountpoint) {
-				return fmt.Errorf("cannot use custom mount points yet, found: %v", mp.Mountpoint)
-			}
+	return nil
+}
+
+// updateFilesystemSizes updates the size of the root filesystem customization
+// based on the minRootSize. The new min size whichever is larger between the
+// existing size and the minRootSize. If the root filesystem is not already
+// configured, a new customization is added.
+func updateFilesystemSizes(fsCustomizations []blueprint.FilesystemCustomization, minRootSize uint64) []blueprint.FilesystemCustomization {
+	updated := make([]blueprint.FilesystemCustomization, len(fsCustomizations), len(fsCustomizations)+1)
+	hasRoot := false
+	for idx, fsc := range fsCustomizations {
+		updated[idx] = fsc
+		if updated[idx].Mountpoint == "/" {
+			updated[idx].MinSize = max(updated[idx].MinSize, minRootSize)
+			hasRoot = true
 		}
-		c.Filesystems = customFS
 	}
+
+	if !hasRoot {
+		// no root customization found: add it
+		updated = append(updated, blueprint.FilesystemCustomization{Mountpoint: "/", MinSize: minRootSize})
+	}
+	return updated
+}
+
+// setRootfsType sets the filesystem type for the mountable entity with target
+// '/'.
+func setRootfsType(pt *disk.PartitionTable, rootfs string) error {
+	rootMountable := pt.FindMountable("/")
+	if rootMountable == nil {
+		return fmt.Errorf("internal partition table does not define a root filesystem")
```

Comment:
*maybe* (I'm really not sure) we should include the incorrect partition table here as well via `%+v` or something to give us a maximum of debug output for this internal error. It may make the error very ugly but maybe it's worth it because the error is supposed to never happen :)

Reply:
Heh, it's a tricky decision isn't it.  Yeah, this "shouldn't happen", but if it does somehow, do we need the full output or do we want it to be readable?  I guess making sure we have all the information in this rare (impossible) event is better.

Ok, you convinced me.  I'll include it.

Reply:
Done.